server/routes/projects.js

```diff
@@ -0,0 +1,55 @@
+const { ObjectId } = require('bson');
+const express = require('express');
+const projectsData = require('../data/projects');
+const {
+	sendErrResp,
+	isValidArray,
+	isValidObj,
+	isValidStr,
+} = require('../utils');
+const { isValidProjectName } = require('../utils/projects');
+const router = express.Router();
+const { authenticateToken } = require('../middleware/auth');
+
+router.route('/project').post(authenticateToken, async (req, res) => {
+	const { user } = req;
+	const {
+		name,
+		description,
+		github,
+		media,
+		technologies,
+		owner,
+		comments,
+		likes,
+	} = req.body;
+	try {
+		// name,description,github,media,technologies,owner,comments,likes
+		// make the validations sent as parameters accurate later
+		name = isValidProjectName(name);
+		description = isValidStr(description);
+		github = isValidStr(github);
+		media = isValidArray(media);
+		technologies = isValidArray(technologies);
+		owner = isValidObj(owner);
+		comments = isValidArray(comments);
+		likes = isValidArray(likes);
```

Comment:
the create object request object will not have a comments and likes array, we will have to initialize them as empty arrays